v0/destinations/one_signal/transform.js

```diff
@@ -0,0 +1,286 @@
+const get = require("get-value");
+const { EventType } = require("../../../constants");
+const {
+  ConfigCategory,
+  mappingConfig,
+  BASE_URL,
+  ENDPOINTS
+} = require("./config");
+const {
+  defaultRequestConfig,
+  getFieldValueFromMessage,
+  constructPayload,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  CustomError,
+  getErrorRespEvents,
+  getSuccessRespEvents,
+  getDestinationExternalID,
+  isDefinedAndNotNullAndNotEmpty,
+  toUnixTimestamp,
+  defaultPutRequestConfig
+} = require("../../util");
+const { populateDeviceType, populateTags } = require("./util");
+
+const responseBuilder = (payload, endpoint, eventType) => {
+  if (payload) {
+    const response = defaultRequestConfig();
+    response.endpoint = `${BASE_URL}${endpoint}`;
+    response.headers = {
+      Accept: "application/json",
+      "Content-Type": "application/json"
+    };
+    if (eventType.toLowerCase() === "identify") {
+      response.method = defaultPostRequestConfig.requestMethod;
+    } else {
+      response.method = defaultPutRequestConfig.requestMethod;
+    }
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+    return response;
+  }
```

Comment:
Throw an error if payload is not populated.

Reply:
Done

Reply:
Done